index.bs

```diff
@@ -0,0 +1,565 @@
+
+1.  Set |this|.[[{{screenState}}]] to {{"unlocked"}}.
+1.  [=Fire an event=] named {{"change"}} at |this|.
+
+<wpt>
+  interceptor.https.html
+</wpt>
+</div>
+
+### {{IdleDetector/onchange}} attribute ### {#api-idledetector-onchange}
+
+<dfn attribute for="IdleDetector">onchange</dfn> is an <a>Event handler IDL
+attribute</a> for the {{change}} event type.
+
+### {{IdleDetector/requestPermission()}} method ### {#api-idledetector-requestpermission}
+
+<div algorithm>
+The <dfn method for=IdleDetector>requestPermission()</dfn> method steps are:
+
+1.  Let |result| be [=a new promise=].
+1.  [=Queue a task=] to run these steps:
+    1.  If the [=relevant global object=] of [=this=] does not have [=transient
+        activation=], [=reject=] |result| with a "{{NotAllowedError}}"
+        {{DOMException}} and abort these steps.
+    1.  Let |permissionState| be the result of [=requesting permission to use=]
+        the [=powerful feature=] named `"idle-detection"`.
+    1.  Resolve |result| with |permissionState|.
+1. Return |result|.
+
+</div>
+
+### {{IdleDetector/start()}} method ### {#api-idledetector-start}
+
+<div algorithm>
+
+The <dfn method for=IdleDetector>start(|options|)</dfn> method, when invoked,
+MUST run these steps:
+
+1.  Let |result| be [=a new promise=].
+1.  If the [=relevant global object=]'s associated {{Document}} is not [=allowed
+    to use=] the [=policy-controlled feature=] named `"idle-detection"`
+    [=reject=] |result| with a "{{NotAllowedError}}" {{DOMException}} and return
+    |result|.
+
+    <wpt>
+      idle-detection-allowed-by-feature-policy-attribute-redirect-on-load.https.sub.html
+      idle-detection-allowed-by-feature-policy-attribute.https.sub.html
+      idle-detection-allowed-by-feature-policy.https.sub.html
+      idle-detection-default-feature-policy.https.sub.html
+      idle-detection-disabled-by-feature-policy.https.sub.html
+    </wpt>
+1.  If |this|.{{IdleDetector/[[state]]}} is not `"stopped"`, [=reject=] |result|
+    with an "{{InvalidStateError}}" {{DOMException}} and return |result|.
+    <wpt>
+      interceptor.https.html
+    </wpt>
+1.  Set |this|.{{IdleDetector/[[state]]}} to `"starting"`.
+1.  If |options|.{{threshold}} is less than 60,000 [=reject=] |result| with
+    {{TypeError}} and return |result|.
+    <wpt>
+      basics.tentative.https.window.js
+    </wpt>
+1.  If |options|.{{signal}} is present, then perform the following sub-steps:
+    1.  If |options|.{{signal}}'s [=AbortSignal/aborted flag=] is set, then
+        [=reject=] |result| with an "{{AbortError}}" {{DOMException}} and return
+        |result|.
+    1.  [=AbortSignal/add|Add the following abort steps=] to
+        |options|.{{signal}}:
+        1. Set |this|.{{IdleDetector/[[state]]}} to `"stopped"`.
+        1. [=Reject=] |result| with an "{{AbortError}}" {{DOMException}}.
+
+    <wpt>
+      interceptor.https.html
+    </wpt>
+1.  [=Queue a task=] to perform the following steps, but [=abort when=]
```

Comment:
This should be "queue a global task on the XXX task source given this's relevant global object to..."

Reply:
Done.